packages/unmock-core/src/service/matcher.ts

```diff
@@ -130,32 +132,28 @@ export class OASMatcher {
         }
       }
     }
-
     const directMatch = paths[reqPath];
     if (directMatch !== undefined) {
       debugLog(`Found direct path match for ${reqPath}`);
-      return reqPath;
+      return { schemaEndpoint: reqPath, normalizedEndpoint: reqPath };
     }
 
     const definedPaths = Object.keys(paths);
     debugLog(`Searching for match for ${reqPath} for ${definedPaths}`);
 
     for (const pathItemKey of Object.keys(paths)) {
       const pathRegex = this.endpointToRegexMapping[pathItemKey];
-      if (pathRegex === undefined) {
-        continue;
-      }
-      if (pathRegex.test(reqPath)) {
-        return pathItemKey;
+      if (XRegExp.test(reqPath, pathRegex)) {
```

Comment:
Figured this can't be undefined as we build `endpointToRegexMapping` for every `Object.keys(paths)`